src/util/Store.js

```diff
@@ -369,45 +368,13 @@ const Store: ModelType<StoreType> = types
   })
   .views((self: StoreType) => ({
     get hasValidParams() {
-      return !!self.agentid && !!self.datasetid && !!self.query
+      return !!self.dataset && (!!self.query || !!self.savedQueryId)
     },
 
     get parsedUrlQuery() {
       return parseParams(self.location.search)
     },
 
-    /*
```

Comment:
`get agentid` and `get datasetid` isn't needed because we're not passing that anymore to chart-builder, we're just passing `dataset` now. 

`get possibleResourceFromQuery` was used only within `get agentid` and `get datasetid` so that was also able to be removed 👍